### DISCUSSION TITLE
Do we need a fancy homepage for https://gplates.github.io/gplately/?

### DISCUSSION THREAD

**Discussion by michaelchin** [GPlates/gplately#213] [Ideas]
Just floating the idea. Currently the index.html was generated by pdoc, which is not fancy and modern looking. Here is an example of good looking homepage https://getbootstrap.com/

**Comment by jcannon-gplates** [comment]
[ReadtheDocs](https://about.readthedocs.com/?ref=readthedocs.com) is pretty good.

I use it for pyBacktrack - there's a little overview [here](https://github.com/EarthByte/pyBacktrack/tree/master/docs#pybacktrack-documentation) about how that's set up - I have a latest dev version and latest release version (but you can have more versions) - and I write the doc pages in reStructuredText (reST) and use numpydoc for the API docstrings (since it's simpler - don't need the power of reST there). I also use it for pyGPlates (but haven't yet set that up to render to the ReadtheDocs website).

**Reply by michaelchin** [reply]
> [ReadtheDocs](https://about.readthedocs.com/?ref=readthedocs.com) is pretty good.
> 
> I use it for pyBacktrack - there's a little overview [here](https://github.com/EarthByte/pyBacktrack/tree/master/docs#pybacktrack-documentation) about how that's set up - I have a latest dev version and latest release version (but you can have more versions) - and I write the doc pages in reStructuredText (reST) and use numpydoc for the API docstrings (since it's simpler - don't need the power of reST there). I also use it for pyGPlates (but haven't yet set that up to render to the ReadtheDocs website).

Yes, I think readthedocs is a good idea. At some point, we should consider migrating to readthedocs.

**Comment by michaelchin** [comment]
Also, a pretty landing page may be nice to have. The pdoc3 itself uses a [separate hand-crafted landing page](https://pdoc3.github.io/pdoc/) which is built with [bulma](https://bulma.io/) CSS.

Anyway, this is just an idea leaping through my mind...

**Reply by jcannon-gplates** [reply]
Just looked at the pdoc3 page you linked. They say...

> Ever found Sphinx too cumbersome to use for the small project on your hands or have repeatedly found reST syntax for links too volatile to touch-type

...where Sphinx and reST is what ReadtheDocs is based on. And pdoc supports...

> Common docstring formats
>
> Out-of-the-box support for markdown, numpydoc, Google-style docstrings, Napoleon, doctests, and some common reST directives.

I haven't used pdoc myself, but if it's easier to use than Sphinx/reST then you may want to continuing using it.  I'll keep using Sphinx for pyGPlates, but only because I've invested in it.

**Comment by brmather** [comment]
I agree it would be good to have a pretty landing page for the GPlately documentation. As you say @michaelchin pdoc3 itself has a fancy landing page. Perhaps we could follow their example...
